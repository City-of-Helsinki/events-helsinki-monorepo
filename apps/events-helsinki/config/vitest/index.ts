export * from './initI18n';
export * from './mockDataUtils';
export * from './test-utils';
export { default as TestProviders } from './TestProviders';
