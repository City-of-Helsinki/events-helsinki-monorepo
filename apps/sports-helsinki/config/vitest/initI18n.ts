export {
  initTestI18n as initI18n,
  initializeI18nWithConfig,
  translations,
} from '@events-helsinki/common-i18n';
