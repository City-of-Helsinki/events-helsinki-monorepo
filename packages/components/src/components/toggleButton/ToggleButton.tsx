import classNames from 'classnames';
import type { MutableRefObject } from 'react';
import React from 'react';

import styles from './toggleButton.module.scss';

type Props = {
  buttonRef?: MutableRefObject<HTMLButtonElement | null>;
  icon?: React.ReactElement;
  isSelected: boolean;
  isExpanded?: boolean;
  onClick: (value: string) => void;
  testId?: string;
  text: string;
  value: string;
} & Pick<
  React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  >,
  'aria-haspopup'
>;

const ToggleButton: React.FC<Props> = ({
  buttonRef,
  icon,
  isSelected,
  onClick,
  testId,
  text,
  value,
  isExpanded,
  'aria-haspopup': ariaHasPopup,
}) => {
  const handleClick = () => {
    onClick(value);
  };
  return (
    <button
      ref={buttonRef}
      data-testid={testId}
      className={classNames(styles.toggleButton, {
        [styles.isSelected]: isSelected,
      })}
      aria-pressed={isSelected}
      aria-expanded={isExpanded}
      aria-haspopup={ariaHasPopup}
      aria-controls={buttonRef?.current?.id}
      aria-label={text}
      onClick={handleClick}
      type="button"
    >
      <div className={styles.innerWrapper}>
        {icon}
        <span>{text}</span>
      </div>
    </button>
  );
};

export default ToggleButton;
