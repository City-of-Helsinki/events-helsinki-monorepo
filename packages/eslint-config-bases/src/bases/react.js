/**
 * Opinionated config base for projects using react.
 * @see https://github.com/City-of-Helsinki/events-helsinki-monorepo/tree/main/packages/eslint-config-bases
 */

import react from 'eslint-plugin-react';
import reactHooks from 'eslint-plugin-react-hooks';

const reactPatterns = {
  files: ['**/*.{js,jsx,ts,tsx}'],
};

export default [
  {
    files: reactPatterns.files,
    plugins: {
      react,
    },
    settings: {
      react: {
        version: 'detect',
      },
    },
    rules: {
      ...react.configs.recommended.rules,
      // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unknown-property.md
      'react/no-unknown-property': ['error', { ignore: ['css'] }],
      // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unescaped-entities.md
      'react/no-unescaped-entities': ['error', { forbid: ['>'] }],
      'react/prop-types': 'off',
      'react/react-in-jsx-scope': 'off',
      // Fine-tune naming convention react typescript jsx (function components)
      // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/naming-convention.md
      '@typescript-eslint/naming-convention': [
        'warn',
        {
          selector: 'variable',
          format: ['camelCase', 'PascalCase', 'UPPER_CASE'],
          leadingUnderscore: 'allow',
        },
        {
          selector: ['function'],
          format: ['camelCase', 'PascalCase'],
        },
        {
          selector: 'parameter',
          format: ['camelCase', 'PascalCase'],
          leadingUnderscore: 'allow',
        },
      ],
    },
  },
  {
    files: reactPatterns.files,
    plugins: {
      'react-hooks': reactHooks,
    },
    rules: {
      ...reactHooks.configs.recommended.rules,
      //   'react-hooks/rules-of-hooks': 'error',
      //   'react-hooks/exhaustive-deps': 'warn',
    },
  },
];
