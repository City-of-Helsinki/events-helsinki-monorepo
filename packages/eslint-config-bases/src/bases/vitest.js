/**
 * Custom config base for projects using vitest.
 * @see https://github.com/City-of-Helsinki/events-helsinki-monorepo/tree/main/packages/eslint-config-bases
 */
import vitest from '@vitest/eslint-plugin';

const vitestPatterns = {
  files: ['**/?(*.)+(test).{js,jsx,ts,tsx}'],
};

export default [
  {
    // Perf: To ensure best performance enable eslint-plugin-vitest for test files only.
    files: vitestPatterns.files,
    plugins: {
      vitest,
    },
    languageOptions: {
      globals: {
        ...vitest.environments.env.globals,
      },
    },
    rules: {
      ...vitest.configs.recommended.rules,
      'vitest/prefer-hooks-in-order': 'error',
      'vitest/prefer-hooks-on-top': 'error',
      'vitest/no-duplicate-hooks': 'error',
      'vitest/no-test-return-statement': 'error',
      'vitest/prefer-strict-equal': 'error',
      'vitest/prefer-to-have-length': 'error',
      'vitest/consistent-test-it': ['error', { fn: 'it' }],
      '@typescript-eslint/no-non-null-assertion': 'off',
      '@typescript-eslint/no-object-literal-type-assertion': 'off',
      '@typescript-eslint/no-empty-function': 'off',
      '@typescript-eslint/no-explicit-any': 'off',
      '@typescript-eslint/ban-ts-comment': 'off',
    },
  },
];
