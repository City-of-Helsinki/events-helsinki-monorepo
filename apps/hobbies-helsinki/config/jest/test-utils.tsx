import type { ApolloCache, InMemoryCache } from '@apollo/client/cache';
import type { MockedResponse } from '@apollo/client/testing';
import type { RenderResult } from '@testing-library/react';
import { act, fireEvent, render } from '@testing-library/react';
import type { NextRouter } from 'next/router';
import Router from 'next/router';
import type { ReactElement } from 'react';
import React from 'react';
import wait from 'waait';

import { hobbiesApolloClient } from '../../src/domain/clients/hobbiesApolloClient';
import TestProviders from './TestProviders';

type CustomRender = {
  (
    ui: React.ReactElement,
    options?: {
      mocks?: MockedResponse[];
      cache?: ApolloCache<Record<string, unknown>> | InMemoryCache;
      routes?: string | string[];
    }
  ): CustomRenderResult;
};

export type CustomRenderResult = RenderResult & { router: NextRouter };

export const arrowUpKeyPressHelper = (): boolean =>
  fireEvent.keyDown(document, { code: 38, key: 'ArrowUp' });

export const arrowDownKeyPressHelper = (): boolean =>
  fireEvent.keyDown(document, { code: 40, key: 'ArrowDown' });

export const enterKeyPressHelper = (): boolean =>
  fireEvent.keyDown(document, { code: 13, key: 'Enter' });

export const escKeyPressHelper = (): boolean =>
  fireEvent.keyDown(document, { code: 27, key: 'Escape' });

export const tabKeyPressHelper = (): boolean =>
  fireEvent.keyDown(document, { code: 9, key: 'Tab' });

const customRender: CustomRender = (
  ui: ReactElement,
  { mocks = [], cache = hobbiesApolloClient.cache, routes = [] } = {}
) => {
  if (routes) {
    if (!Array.isArray(routes)) {
      routes = [routes];
    }
    routes.forEach((route) => Router.push(route));
  }

  // set locales so that the routing tests for default locale worked
  Router.locales = ['fi', 'en', 'sv'];

  const renderResult = render(ui, {
    wrapper: ({ children }) => (
      <TestProviders mocks={mocks} router={Router} cache={cache}>
        {children}
      </TestProviders>
    ),
  });
  return {
    ...renderResult,
    router: Router,
  };
};

// eslint-disable-next-line testing-library/no-unnecessary-act
const actWait = (amount?: number): Promise<void> => act(() => wait(amount));

// eslint-disable-next-line import/export
export { actWait, customRender as render };

// re-export everything
// eslint-disable-next-line import/export
export * from '@testing-library/react';
export { render as defaultRender } from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
