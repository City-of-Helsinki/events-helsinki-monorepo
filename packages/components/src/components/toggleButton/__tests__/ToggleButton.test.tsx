import userEvent from '@testing-library/user-event';
import * as React from 'react';
import { vi } from 'vitest';
import { render } from '@/test-utils';
import ToggleButton from '../ToggleButton';

const defaultProps = {
  isSelected: true,
  onClick: vi.fn(),
  text: 'Test button',
  value: 'test',
};

it('matches snapshot', () => {
  const { container } = render(<ToggleButton {...defaultProps} />);

  expect(container.firstChild).toMatchSnapshot();
});

it('should call onClick', async () => {
  const onClick = vi.fn();
  const { container } = render(
    <ToggleButton {...defaultProps} onClick={onClick} />
  );

  await userEvent.click(container.firstChild as HTMLElement);
  expect(onClick).toHaveBeenCalledTimes(1);
});
