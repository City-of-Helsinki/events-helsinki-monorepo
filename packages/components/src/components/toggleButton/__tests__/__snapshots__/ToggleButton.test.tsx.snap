// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  aria-pressed="true"
  class="_toggleButton_9d9d09 _isSelected_9d9d09"
  type="button"
>
  <div
    class="_innerWrapper_9d9d09"
  >
    <span>
      Test button
    </span>
  </div>
</button>
`;
